Source/ACE/Network/NetworkSession.cs

```diff
@@ -0,0 +1,359 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using ACE.Network.GameMessages;
+using ACE.Network.Managers;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Net.Sockets;
+using ACE.Network.Handlers;
+
+namespace ACE.Network
+{
+    public class NetworkSession
+    {
+        private const int minimumTimeBetweenBundles = 5; // 5ms
+        private const int timeBetweenTimeSync = 20000; // 20s
+        private const int timeBetweenAck = 2000; // 2s
+
+        private ConnectionType connectionType;
+        private Session session;
+        private NetworkBundle currentBundle;
+        private DateTime nextResync;
+        private DateTime nextAck;
+        private DateTime nextSend;
+        private bool sendAck = false;
+        private bool sendResync = false;
+        private uint lastReceivedSequence = 0;
+        private ConcurrentDictionary<uint /*seq*/, ServerPacket> CachedPackets { get; } = new ConcurrentDictionary<uint, ServerPacket>();
+        private ConcurrentQueue<ServerPacket> PacketQueue { get; } = new ConcurrentQueue<ServerPacket>();
+
+        public SessionConnectionData ConnectionData { get; private set; }
+
+        public NetworkSession(Session session, ConnectionType connType)
+        {
+            this.session = session;
+            this.connectionType = connType;
+            ConnectionData  = new SessionConnectionData(connType);
+            currentBundle = new NetworkBundle();
+            nextSend = DateTime.Now;
```

Comment:
I did a global replace for Every DateTime.Now with DateTime.UtcNow and it seems to work ok.